pkg/noderesourcetopology/resourcecounter.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+type resourceCounter v1.ResourceList
```

Comment:
Is the purpose of aliasing v1.ResourceList to come up with a dedicated Add() and Sub() functions? If so, I think we can add a stateless function like:

- func Add(a v1.ResourceList, b v1.ResourceList) v1.ResourceList
- func Sub(a v1.ResourceList, b v1.ResourceList) v1.ResourceList

So that we can eliminating introducing this alias.

Reply:
sure, I'm fine with this and I actually like more the free function approach.